File: VB/MyXPOClassLibrary/PersistentObject1.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Namespace MyXPOClassLibrary

	Public Class PersistentObject1
		Inherits XPObject
		'public PersistentObject1()
		'    : base() {
		'    // This constructor is used when an object is loaded from a persistent storage.
		'    // Do not place any code here.
		'}

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
			' This constructor is used when an object is loaded from a persistent storage.
			' Do not place any code here.
		End Sub

		Public Overrides Sub AfterConstruction()
			MyBase.AfterConstruction()
			' Place here your initialization code.
		End Sub
		Private _stringPropertyName As String
		Public Property StringPropertyName() As String
			Get
				Return _stringPropertyName
			End Get
			Set(ByVal value As String)
				SetPropertyValue("StringPropertyName", _stringPropertyName, value)
			End Set
		End Property
		Private _intPropertyName As Integer
		Public Property IntPropertyName() As Integer
			Get
				Return _intPropertyName
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("IntPropertyName", _intPropertyName, value)
			End Set
		End Property
	End Class
End Namespace

File: CS/CustomizeXPOModel/MyXPOClassLibrary/PersistentObject1.cs
using System;
using System.ComponentModel;
using DevExpress.Xpo;

namespace MyXPOClassLibrary {

    public class PersistentObject1 : XPObject {
        public PersistentObject1(Session session)
            : base(session) {
            // This constructor is used when an object is loaded from a persistent storage.
            // Do not place any code here.
        }

        public override void AfterConstruction() {
            base.AfterConstruction();
            // Place here your initialization code.
        }
        private string _stringPropertyName;
        public string StringPropertyName {
            get { return _stringPropertyName; }
            set { SetPropertyValue(nameof(StringPropertyName), ref _stringPropertyName, value); }
        }
        private int _intPropertyName;
        public int IntPropertyName {
            get { return _intPropertyName; }
            set { SetPropertyValue(nameof(IntPropertyName), ref _intPropertyName, value); }
        }
    }
}